contracts/deposit-contracts/contract/src/main.rs

```diff
@@ -0,0 +1,214 @@
+#![no_std]
+#![no_main]
+extern crate alloc;
+use alloc::{collections::BTreeMap, string::ToString, vec, vec::Vec};
+use casper_contract::{
+    contract_api::{runtime, storage, system},
+    unwrap_or_revert::UnwrapOrRevert,
+};
+use casper_types::{
+    bytesrepr::ToBytes, contracts::NamedKeys, runtime_args, AccessRights, ApiError, CLValue,
+    EntryPoints, Key, RuntimeArgs, URef, U512,
+};
+mod constants;
+use constants::{
+    ADMIN_LIST, KAIROS_DEPOSIT_CONTRACT, KAIROS_DEPOSIT_CONTRACT_NAME,
+    KAIROS_DEPOSIT_CONTRACT_PACKAGE, KAIROS_DEPOSIT_EVENT_DICT, KAIROS_DEPOSIT_PURSE,
+    KAIROS_LAST_PROCESSED_DEPOSIT_COUNTER, KAIROS_MOST_RECENT_DEPOSIT_COUNTER, RUNTIME_ARG_AMOUNT,
+    RUNTIME_ARG_DEST_PURSE, RUNTIME_ARG_TEMP_PURSE, SECURITY_BADGES,
+};
+mod detail;
+use detail::{get_immediate_caller, get_optional_named_arg_with_user_errors};
+mod error;
+use error::DepositError;
+mod security;
+use security::{sec_check, SecurityBadge};
+mod entry_points;
+
+use contract_types::Deposit;
+
+#[no_mangle]
+pub extern "C" fn init() {
+    if runtime::get_key(KAIROS_DEPOSIT_PURSE).is_some() {
+        runtime::revert(DepositError::AlreadyInitialized);
```

Comment:
Will this return or abort the execution of `init`?

Reply:
Revert aborts.